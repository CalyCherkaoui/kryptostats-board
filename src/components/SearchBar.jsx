/* eslint-disable react/prop-types */
import React, { useState } from 'react';
// import PropTypes from 'prop-types';

const SearchBar = (props) => {
  const [search, setSearch] = useState('');

  // const goToSerchedCoin = () => {
  //   props.history.push(`/coininfo/${search}`);
  // };
  return (
    <div className="search_bar_container">
      <input type="text" onChange={(e) => setSearch(e.target.value)} />
      <button type="button" onClick={() => props.history.push(`/coininfo/${search}`)}>Search</button>
    </div>
  );
};

// SearchBar.propTypes = {
//   history: PropTypes.arrayOf.isRequired,
// };

export default SearchBar;
