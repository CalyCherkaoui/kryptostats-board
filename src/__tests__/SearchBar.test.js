import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import SearchBar from '../components/SearchBar';

describe(
  'Renders correctly The navigation bar component',
  () => {
    it(
      'Matches LocalCurrencyFilter snapshot',
      () => {
        const rendered = renderer.create(
          <Provider store={store}>
            <Router>
              <SearchBar />
            </Router>
          </Provider>,
        );
        expect(rendered).toMatchSnapshot();
      },
    );
  },
);
