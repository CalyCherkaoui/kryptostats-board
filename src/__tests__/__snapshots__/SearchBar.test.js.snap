// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly The navigation bar component Matches LocalCurrencyFilter snapshot 1`] = `
<div
  className="search_bar_container"
>
  <input
    className="search_bar_input normal_typography"
    onChange={[Function]}
    placeholder="Search Crypto"
    type="text"
  />
  <button
    className="search_bar_button"
    onClick={[Function]}
    type="button"
  >
    Search
  </button>
</div>
`;
